code/game/turfs/floor.dm

```diff
@@ -1,525 +1,160 @@
-	else
-		icon_state = "floor"
-		icon_regular_floor = icon_state
-	update_icon()
-	levelupdate()
-
-
-//This proc will make the turf a light floor tile. The expected argument is the tile to make the turf with
-//If none is given it will make a new object. dropping or unequipping must be handled before or after calling
-//this proc.
-/turf/open/floor/proc/make_light_floor(obj/item/stack/tile/light/newtile)
-	broken = 0
-	burnt = 0
-	intact_tile = 1
-	if(!istype(newtile))
-		newtile = new
-
-	floor_tile = newtile
-	update_icon()
-	levelupdate()
-
-
-//This proc will make a turf into a grass patch. Fun eh? Insert the grass tile to be used as the argument
-//If no argument is given a new one will be made.
-/turf/open/floor/proc/make_grass_floor(obj/item/stack/tile/grass/T = null)
-	broken = 0
-	burnt = 0
-	intact_tile = 1
-	if(T)
-		if(istype(T,/obj/item/stack/tile/grass))
-			floor_tile = T
-			update_icon()
-			levelupdate()
-			return
-	//If you gave a valid parameter, it won't get thisf ar.
-	floor_tile = new/obj/item/stack/tile/grass
-
 	update_icon()
-	levelupdate()
 
-//This proc will make a turf into a wood floor. Fun eh? Insert the wood tile to be used as the argument
-//If no argument is given a new one will be made. SOMEONE FIX THIS COPYPASTE BULLSHIT!!
-/turf/open/floor/proc/make_wood_floor(obj/item/stack/tile/wood/T = null)
-	broken = 0
-	burnt = 0
-	intact_tile = 1
-	if(T)
-		if(istype(T,/obj/item/stack/tile/wood))
-			floor_tile = T
-			update_icon()
-			levelupdate()
-			return
-	//If you gave a valid parameter, it won't get thisf ar.
-	floor_tile = new/obj/item/stack/tile/wood
-
-	update_icon()
-	levelupdate()
-
-//This proc will make a turf into a carpet floor. Fun eh? Insert the carpet tile to be used as the argument
-//If no argument is given a new one will be made.
-/turf/open/floor/proc/make_carpet_floor(obj/item/stack/tile/carpet/T = null)
-	broken = 0
-	burnt = 0
-	intact_tile = 1
-	if(T)
-		if(istype(T,/obj/item/stack/tile/carpet))
-			floor_tile = T
-			update_icon()
-			levelupdate()
-			return
-	//If you gave a valid parameter, it won't get thisf ar.
-	floor_tile = new/obj/item/stack/tile/carpet
-
-	update_icon()
-	levelupdate()
+/// Things seem to rely on this actually returning plating. Override it if you have other baseturfs.
+/turf/open/floor/proc/make_plating()
+	return ScrapeAway()
 
-/turf/open/floor/attackby(obj/item/I, mob/user, params)
+/turf/open/floor/attackby(obj/item/object, mob/living/user, params)
+	if(!object || !user)
+		return TRUE
 	. = ..()
-
-	if(hull_floor) //no interaction for hulls
+	if(.)
+		return .
+	if(floor_tile && istype(object, /obj/item/stack/tile))
+		try_replace_tile(object, user, params)
+		return TRUE
+	return FALSE
+
+///swaps out an existing floor tile with another one
+/turf/open/floor/proc/try_replace_tile(obj/item/stack/tile/T, mob/user, params)
+	if(T.turf_type == type)
 		return
+	var/obj/item/tool/crowbar/CB = user.is_holding_item_of_type(/obj/item/tool/crowbar)
```

Comment:
This proc returns true/false, not the item. Also letter vars.

Reply:
done